### DISCUSSION TITLE
RTKQ - How can I always fetch on first render?

### DISCUSSION THREAD

**Discussion by LoyalPotato** [reduxjs/redux-toolkit#3007] [Q&A] [answered]
Hello,

Right now I have an endpoint defined with `queryFn` and uses the `baseQuery` to make the request.
I'm not providing any tags for this endpoint as well.
But when checking the network tab for requests, it's only making a first one request and then on subsequent renders, such as going to another page that is using the same query hook, but it's a new render, it's not making the request.

Why's that? How can I make sure that on every first render, that the request will be made?

**Comment by markerikson** [comment, marked as answer]
@LoyalPotato : that's one of the main purposes of RTKQ - to reuse cached data _without_ having to re-fetch it every time.

If you really want to refetch when a different component mounts, use the [`refetchOnMountOrArgChange`](https://redux-toolkit.js.org/rtk-query/api/createApi#refetchonmountorargchange) in either `createApi` or the hook call itself.

**Reply by LoyalPotato** [reply]
Yeah I know and I do use that feature in the rest of the endpoints :)
It was just one endpoint that I defined in an api that needed this behaviour.
I ended up using the `forceRefetch: true`, which is also mentioned in the link you sent.

Thank you for your help!
